### PR TITLE
vault deployment event emitter

### PR DESCRIPTION
**Pull request by Spablob**

### CODE REVIEW

File: contracts/utils/VaultDeploymentHelper.sol [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+pragma solidity =0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract VaultDeploymentHelper is Ownable {
+    /**
+     * @notice Emitted when a new vault is deployed
+     * @param vaultAddress the address of the new vault
+     */
+    event NewVault(address vaultAddress);
```
Comment (chuddster) [inline contracts/utils/VaultDeploymentHelper.sol:11]: adjust event for vault type (earn/vip/normal/treasury)

File: contracts/utils/VaultDeploymentHelper.sol [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (chuddster) [inline contracts/utils/VaultDeploymentHelper.sol:12]: 1. maybe we can also maintain an **array** of all the vault addresses we have and just update it. can be fixed size array + private counter we maintain. 
2. maybe maintain mapping from vault address => enum type

File: contracts/utils/VaultDeploymentHelper.sol [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    /**
```
Comment (chuddster) [inline contracts/utils/VaultDeploymentHelper.sol:13]: add constructor where we do bulk update of array for all current vaults we have

File: contracts/utils/VaultDeploymentHelper.sol (hunk 1 above)
Comment (Spablob) [reply contracts/utils/VaultDeploymentHelper.sol]: Addressed

File: contracts/utils/VaultDeploymentHelper.sol (hunk 2 above)
Comment (Spablob) [reply contracts/utils/VaultDeploymentHelper.sol]: Added mapping enum type => array of all vaults with that type

Currently, the contract has been renamed to `VaultDeploymentEventEmitter`. Should we rename it to "AddressBook" or keep it as it is?

File: contracts/utils/VaultDeploymentHelper.sol (hunk 3 above)
Comment (Spablob) [reply contracts/utils/VaultDeploymentHelper.sol]: Addressed

File: scripts/deploy/20_wbtc_call_theta_vault_swap.ts [hunk 4]
```diff
@@ -31,6 +31,7 @@ const main = async ({
   );
 
   const manualVolOracle = await deployments.get("ManualVolOracle");
+  const vaultDeploymentEventEmitter = await deployments.get("VaultDeploymentEventEmitter");
```
Comment (Spablob) [inline scripts/deploy/20_wbtc_call_theta_vault_swap.ts:34]: deployment pending

Comment (Spablob) [reply scripts/deploy/20_wbtc_call_theta_vault_swap.ts:34]: deployment completed

File: contracts/utils/VaultDeploymentHelper.sol (hunk 2 above)
Comment (chuddster) [reply contracts/utils/VaultDeploymentHelper.sol]: Lets change to AddressBook since it serves that purpose as well - emitter can be secondary

**Review by chuddster — APPROVED** [review]

**Pull request merged**